collection.go

```diff
@@ -471,6 +471,15 @@ func (cl *collectionLoader) loadMap(mapName string) (*Map, error) {
 		return nil, fmt.Errorf("map %s: %w", mapName, err)
 	}
 
+	// Finalize 'scalar' maps that don't refer to any other eBPF resources
+	// potentially pending creation. This is needed for frozen maps like .rodata
+	// that need to be finalized before invoking the verifier.
+	if !mapSpec.Type.canStoreMapOrProgram() {
+		if err := m.finalize(mapSpec); err != nil {
```

Comment:
Can you remind me why we can't populate deferred maps at this point as well?

Reply:
Map fds need to be relocated into programs before a program can be loaded, but we can't populate a prog array without creating programs first. We've split up this process into multiple stages to keep things clear. Technically, we only need to defer populating prog arrays, but we can refactor this again later if new requirements pop up.